### DISCUSSION TITLE
Kata 4, Transformar cadenas

### DISCUSSION THREAD

**Discussion by AndresAVargas** [LaunchX-InnovaccionVirtual/CursoIntroPython#49] [Q&A]
Compañeros no se si puedan ayudarme, según yo tendria que salir el numero de la temperatura pero solo me sale la palalabra :C
![image](https://user-images.githubusercontent.com/88987587/153290995-28a6044d-1774-4f24-a740-795bb11ef385.png)

**Comment by AzaelBarbosa** [comment]
Hola amigo lo estas definiendo mal en el **REPLACE** primero va la palabra o letra que va a buscar y despues por la que lo reemplazara:

,replace("caracter a buscar", "caracter por el que reemplazara")

`for txt in txt_partes:
    for clave in txt_clave:
        if clave in txt:
            print(txt.replace(' C', ' Celsius'))
            break`

ESpero haberte sido de ayuda :)

**Reply by AndresAVargas** [reply]
en ambos cambie el codigo y en vez de 'sentece' cambie por como defini el texto y sigo igual
![image](https://user-images.githubusercontent.com/88987587/153302822-cd4f67df-8a56-4a8f-9cb6-1bfe5ad88dd2.png)

**Reply by AzaelBarbosa** [reply]
Valida bien tus variables amigo porque veo que tienes definida solo una palabra y no puedo ver el resto del codigo, te dejo mi ejemplo:

```python
text = """Interesting facts about the Moon. The Moon is Earth's only satellite. There are several interesting facts about the Moon and how it affects life here on Earth. 
On average, the Moon moves 4cm away from the Earth every year. This yearly drift is not significant enough to cause immediate effects on Earth. The highest daylight temperature of the Moon is 127 C."""

txt_partes = text.split(". ")

txt_clave = ["average", "temperature", "distance"] #Arreglo

for txt in txt_partes:
    for clave in txt_clave:
        if clave in txt:
            print(txt)
            break

for txt in txt_partes:
    for clave in txt_clave:
        if clave in txt:
            print(txt.replace(' C', ' Celsius'))
            break
```

**Reply by PlayerOneV** [reply]
@AzaelBarbosa  buenas noches, me podrías decir por favor como pusiste el código aqui en git con el formato, la opción de copiar, etc.

**Reply by AzaelBarbosa** [reply]
Hola amigo, claro tienes que encerrar el código en 3 apostrofes

--> ```

'''(Tipo codigo: Java,Python,Javascript,etc)  TU CODIGO AQUI 

'''

Ejemplo:

'''python 
  text= "Hola mundo" 
'''

 ```python 
  text= "Hola mundo"
```

En el ejemplo no te aparece porque son comillas simples solo reemplázalos por apostrofes :)

Te dejo el link de la documentación para apoyo :)
https://docs.github.com/es/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by AndresAVargas** [reply]
Esta raro deja la checo

**Reply by AndresAVargas** [reply]
> Valida bien tus variables amigo porque veo que tienes definida solo una palabra y no puedo ver el resto del codigo, te dejo mi ejemplo:
> 
> ```python
> text = """Interesting facts about the Moon. The Moon is Earth's only satellite. There are several interesting facts about the Moon and how it affects life here on Earth. 
> On average, the Moon moves 4cm away from the Earth every year. This yearly drift is not significant enough to cause immediate effects on Earth. The highest daylight temperature of the Moon is 127 C."""
> 
> txt_partes = text.split(". ")
> 
> txt_clave = ["average", "temperature", "distance"] #Arreglo
> 
> for txt in txt_partes:
>     for clave in txt_clave:
>         if clave in txt:
>             print(txt)
>             break
> 
> for txt in txt_partes:
>     for clave in txt_clave:
>         if clave in txt:
>             print(txt.replace(' C', ' Celsius'))
>             break
> ```

compañero no se si me puedas orientar, ya cheque y segun yo deberia de funcionar y ahora ni texto me imprime, 
![image](https://user-images.githubusercontent.com/88987587/153329281-9649ad67-30dd-4790-8023-51b1a4fd1531.png)
![image](https://user-images.githubusercontent.com/88987587/153329301-c0911da5-8721-4ae8-b8d7-0a262c2d5c49.png)
![image](https://user-images.githubusercontent.com/88987587/153329317-c4a6d4cc-2dce-40c1-91cd-be3338c4ac93.png)
y como te comento no me imprime nada jaja

**Reply by PlayerOneV** [reply]
@AndresAVargas  en tus bucles for estas usando text y esa variable es la que contiene todo el párrafo de texto. Intenta cambiando en los bucles la palabra text por part, tambien en el condicional.

**Reply by AndresAVargas** [reply]
osea text en parts y parts en text :C no entendi soy nuevo,
edit: nope esto no me funciono xD

**Reply by PlayerOneV** [reply]
Te comparto como lo resolví yo, lo único que se me ocurre es que estas usando text en tus bucles y allí declaraste todo el párrafo, cambia text por otra palabra en los for, if y print. Y otra cosa que note es que en el if te aparece azul el "in" mientras que en los for no...
![Captura de pantalla (29)](https://user-images.githubusercontent.com/75816087/153332022-4dbdd320-61c9-4ada-9862-34a12a71b2f0.png)

**Reply by ViktorVizcarra** [reply]
> en ambos cambie el codigo y en vez de 'sentece' cambie por como defini el texto y sigo igual ![image](https://user-images.githubusercontent.com/88987587/153302822-cd4f67df-8a56-4a8f-9cb6-1bfe5ad88dd2.png)

yo lo hice asi y funciona:

# Ciclo para cambiar C a Celsius
for palabra in text.split():
    if palabra == 'C.':
        print(text.replace('C.','Celsius.'))

**Reply by AzaelBarbosa** [reply]
Excelente amigo, no importa el metodo mientras funcione y hayas entendido el código :D

**Reply by AndresAVargas** [reply]
> > en ambos cambie el codigo y en vez de 'sentece' cambie por como defini el texto y sigo igual ![image](https://user-images.githubusercontent.com/88987587/153302822-cd4f67df-8a56-4a8f-9cb6-1bfe5ad88dd2.png)
> 
> yo lo hice asi y funciona:
> 
> # Ciclo para cambiar C a Celsius
> for palabra in text.split():
>     if palabra == 'C.':
>         print(text.replace('C.','Celsius.'))

Si ya cambie el codigo y de echo ya jalo pero no se si deba dar todo el texto por que a mi solo me da la cadena de la temperatura y supuse que estaba bien

**Reply by YomairaBarrera** [reply]
Gracias Muchachos me salvaron gracias a sus explicaciones al fin le entendí y lo pude hacer
